### DISCUSSION TITLE
Support for `ssl_certificate_password` to allow password to protected certificate

### DISCUSSION THREAD

**Discussion by lukas0krupa** [nginx-proxy/nginx-proxy#1893] [Q&A] [answered]
Hi All,

I am just starting to use your version of Nginx, was a little bit unsucessful struggle with configuring Nginx myself.

However, so far I missing an option to supply `ssl_certificate_password` ([as documented here](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_password_file)), any changes that can be implemented? (conditionally if `<docmain>.pass` is present.

Thanks,
Lukas

PS: I did search, but I haven't find any related issue, so apologies if this is already opened.

**Comment by cuzzo333** [comment]
This is a feature my team would like as well.  We deploy to customers who have different security requirements determined by their organization and password protected key files are one of those.

**Comment by ykammeyer** [comment, marked as answer]
I also have to provide passwords for keys. My solution was to add this entry to a vhost configuration. Basically, I added/inserted the `ssl_password_file <path-to-password-file>;` entry to the `/etc/nginx/vhost.d/my.host` file. Maybe this was not an option back in 2020, but it is one currently. On how to setup per host configuration, I just followed the nginx-proxy documentation. Maybe there is another way, that I missed?
